src/Approximations/overapproximate_expmap.jl

```diff
@@ -0,0 +1,249 @@
+"""
+    _compute_inner_powers(A::MatrixZonotope, P::S,
+                               k::Int) where {S<:Union{SparsePolynomialZonotope,
+                                                       AbstractZonotope}}
+    
+Compute the first `k` inner powers of the matrix zonotope `A` applied to the set `P`.
+
+This function returns a vector of overapproximated sets of the form:
+
+```math
+    (1/n!) * \\mathcal{A}^n * P
+```
+for `n = 0` to `k`.
+"""
+function _compute_inner_powers(A::MatrixZonotope, P::S,
+                               k::Int) where {S<:Union{SparsePolynomialZonotope,
+                                                       AbstractZonotope}}
+    T = typeof(P)
+    powers = Vector{T}(undef, k + 1)
+    invfact = 1
+    powers[1] = P
+
+    @inbounds for n in 1:k
+        invfact /= n
+        term = overapproximate(scale(invfact, A) * powers[n], T)
+        powers[n + 1] = term
+    end
+    return powers
+end
+
+"""
+    _compute_outer_powers(B::MatrixZonotope, in_powers::Vector{S},
+                               k::Int) where {S<:Union{SparsePolynomialZonotope,
+                                                       AbstractZonotope}}
+
+Apply `B` repeatedly to each element in `in_powers`, approximating:
+
+```math
+B^n * (A^n * P)
+```
+
+for `n = 0` to `k`.
+"""
+function _compute_outer_powers(B::MatrixZonotope, in_powers::Vector{S},
+                               k::Int) where {S<:Union{SparsePolynomialZonotope,
+                                                       AbstractZonotope}}
+    out_powers = similar(in_powers)
+    out_powers[1] = in_powers[1]
+
+    @inbounds for n in 2:(k + 1)
+        term = in_powers[n]
+        for _ in 1:n
+            term = overapproximate(B * term, S)
```

Comment:
Should this not be running once shorter (since `n = i+1`)?
```suggestion
        for _ in 1:(n-1)
            term = overapproximate(B * term, S)
```